apps/src/redux/lti/ltiFeedbackReducer.ts

```diff
@@ -0,0 +1,113 @@
+import {Dispatch} from 'react';
+
+import {getAuthenticityToken} from '@cdo/apps/util/AuthenticityTokenStore';
+
+const LTI_FEEDBACK_URL = '/lti/v1/feedback';
+const LTI_FEEDBACK_FETCH_REQUEST = 'lti/feedback/fetchRequest';
+const LTI_FEEDBACK_FETCH_SUCCESS = 'lti/feedback/fetchSuccess';
+const LTI_FEEDBACK_FETCH_FAILURE = 'lti/feedback/fetchFailure';
+const LTI_FEEDBACK_CREATE_REQUEST = 'lti/feedback/createRequest';
+const LTI_FEEDBACK_CREATE_SUCCESS = 'lti/feedback/createSuccess';
+const LTI_FEEDBACK_CREATE_FAILURE = 'lti/feedback/createFailure';
+
+type LtiFeedbackAction =
+  | {type: typeof LTI_FEEDBACK_FETCH_REQUEST}
+  | {type: typeof LTI_FEEDBACK_FETCH_SUCCESS; ltiFeedback: LtiFeedback | null}
+  | {type: typeof LTI_FEEDBACK_FETCH_FAILURE; error: string}
+  | {type: typeof LTI_FEEDBACK_CREATE_REQUEST}
+  | {type: typeof LTI_FEEDBACK_CREATE_SUCCESS; ltiFeedback: LtiFeedback}
+  | {type: typeof LTI_FEEDBACK_CREATE_FAILURE; error: string};
+
+interface LtiFeedback {
+  id?: number;
+  satisfied: boolean;
+  locale?: string;
+  early_access?: boolean;
+}
+
+interface LtiFeedbackState {
+  isLoading: boolean;
+  ltiFeedback?: LtiFeedback | null;
+  error?: string;
+}
+
+export function ltiFeedbackReducer(
+  state: LtiFeedbackState,
+  action: LtiFeedbackAction
+): LtiFeedbackState {
+  switch (action.type) {
+    case LTI_FEEDBACK_FETCH_REQUEST:
+      return {...state, isLoading: true};
+    case LTI_FEEDBACK_FETCH_SUCCESS:
+      return {...state, isLoading: false, ltiFeedback: action.ltiFeedback};
+    case LTI_FEEDBACK_FETCH_FAILURE:
+      return {...state, isLoading: false, error: action.error};
+    case LTI_FEEDBACK_CREATE_REQUEST:
+      return {...state, isLoading: true};
+    case LTI_FEEDBACK_CREATE_SUCCESS:
+      return {...state, isLoading: false, ltiFeedback: action.ltiFeedback};
+    case LTI_FEEDBACK_CREATE_FAILURE:
+      return {...state, isLoading: false, error: action.error};
+    default:
+      return state;
+  }
+}
+
+export const fetchLtiFeedback = (dispatch: Dispatch<LtiFeedbackAction>) => {
```

Comment:
Is it possible to use an async function and try/catch with `await` instead of chaining `.then`s?

Reply:
It's a reducer that returns data via a dispatcher and not directly via Promise:
https://github.com/code-dot-org/code-dot-org/blob/f9fb91cc6df63eb7d0c780d96b1555f389214722/apps/src/lib/ui/lti/feedback/LtiFeedbackBanner.tsx#L37-L42